keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -715,45 +722,45 @@ spec:
     - protocol: TCP
       port: 80
       targetPort: 9376
-  preferDualStack: false # defaulted
-  ipFamilies:            # set by user
+  ipFamilyPolicy: SingleStack # defaulted
+  ipFamilies:                 # set by user
     - IPv6
-  clusterIP: 2001::1     # allocated
+  clusterIP: 2001::1          # allocated
   clusterIPs:
     - 2001::1
 ```
 
-> It is assumed that any service created with `spec.preferDualStack: false` is
+> It is assumed that any service created with `spec.ipFamilyPolicy: nil` is
 > single-stack. Users can change this behavior using admission control web
 > hooks, if they want to default services to dual-stack.
 
 ##### Creating a New Dual-Stack Service
 
 Users can create-dual stack services according to the following methods (in
 increasing specificity):
-- If the user prefers dual stack (if available, service creation will not fail if
+- If the user *prefers* dual stack (if available, service creation will not fail if
   the cluster is not configured for dual stack) then they can do one of the
   following:
-  1. Set `spec.preferDualStack` to `true` and do not set `spec.ipFamilies` or
+  1. Set `spec.ipFamilyPolicy` to `PreferDualStack` and do not set `spec.ipFamilies` or
      `spec.clusterIPs`. The apiserver will set `spec.ipFamilies` according to
      how the cluster is configured, and will allocate IPs according to those
      families.
-  2. Set `spec.preferDualStack` to `true`, set `spec.ipFamilies` to one IP
+  2. Set `spec.ipFamilyPolicy` to `PreferDualStack`, set `spec.ipFamilies` to one IP
      family, and do not set `spec.clusterIPs`. The apiserver will set
      `spec.ipFamilies` to the requested family if it is available (otherwise it
      will fail) and the alternate family if dual-stack is configured, and will
      allocate IPs according to those families.
-  3. Set `spec.preferDualStack` to `true`, either do not set `spec.ipFamilies`
+  3. Set `spec.ipFamilyPolicy` to `PreferDualStack`, either do not set `spec.ipFamilies`
      or set it to one IP family, and set `spec.clusterIPs` to one IP.
      The apiserver will default `spec.ipFamilies` to the family of
      `clusterIPs[0]` and the alternate family if dual-stack is configured, will
      reserve the specified IP if possible, and will allocate a second IP of the
      alternate family if dual-stack is configured.
 - If the user *requires* dual-stack service (service creation will fail if cluster
   is not configured for dual-stack) then they can do one of the following:
-  1. Either don't set `spec.preferDualStack` or set it to `true`, and set
+  1. Either don't set `spec.ipFamilyPolicy` or set it to `RequireDualStack`, and set
      `spec.ipFamilies` to the two IP families in any order.
-  2. Either don't set `spec.preferDualStack` or set it to `true`, and set
+  2. Either don't set `spec.ipFamilyPolicy` or set it to `RequireDualStack`, and set
      `spec.clusterIPs` to any two IPs of different families, in any order.
```

Comment:
There's a third way now (which should probably come first in the list), which is to set `spec.ipFamilyPolicy` to `RequireDualStack` and not set `spec.ipFamilies` or `spec.clusterIPs`